tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/chunk/ChunkReader.java

```diff
@@ -131,10 +132,25 @@ private void skipBytesInStreamByLength(long length) {
   }
 
   public boolean pageSatisfied(PageHeader pageHeader) {
-    if (pageHeader.getEndTime() <= deletedAt) {
-      return false;
-    } else if (pageHeader.getStartTime() <= deletedAt) {
-      pageHeader.setModified(true);
+    long lower = pageHeader.getStartTime();
+    long upper = pageHeader.getEndTime();
+    // deleteIntervalList is sorted in terms of startTime
+    if (deleteIntervalList != null) {
+      for (Pair<Long, Long> range : deleteIntervalList) {
+        if (upper < range.left) {
+          break;
+        }
+        if (range.left <= lower && lower <= range.right) {
+          pageHeader.setModified(true);
+          if (upper <= range.right) {
+            return true;
```

Comment:
if the page is totally deleted, this page is not satisfied, return false.

Reply:
Fixed